crates/bevy_render/src/view/window.rs

```diff
@@ -239,11 +239,22 @@ pub fn prepare_windows(
             .flags;
 
         if !sample_flags.sample_count_supported(msaa.samples()) {
-            bevy_log::warn!(
-                "MSAA {}x is not supported on this device. Falling back to disabling MSAA.",
-                msaa.samples(),
-            );
-            *msaa = Msaa::Off;
+            if sample_flags.sample_count_supported(Msaa::default().samples()) {
```

Comment:
Rather than two different log statements in and f, use the if statement to determine the fallback sample count, and then set that once and use one log statement. That will be cleaner.

Pseudo code:
```rust
let fallback_sample = if default_supported() { default } else { off };
log("Not supported, falling back to {fallback_sample}");
*msaa = fallback_sample;
```